### DISCUSSION TITLE
Uncaught ReferenceError: global is not defined

### DISCUSSION THREAD

**Discussion by mohan-reviewer1** [webpack/webpack#19053] [Q&A]
We are trying to update node from 12 to 20 and electron version from 11 to 31.

I have an **Electron** based application. we do **webpack**  in the render process. It throws error when I run the app after update
**Uncaught ReferenceError: global is not defined**

**If the current behavior is a bug, please provide the steps to reproduce.**

**What is the expected behavior?**
It should not throw this error when I run the application

Other relevant information:
**here is the Error:**
  jsonp chunk loading:79 Uncaught ReferenceError: global is not defined
    at jsonp chunk loading:79:1
    at jsonp chunk loading:575:1
    at startup:7:1
Note: Since it is an electron app:
config.target = "electron-renderer";

we have config-overrides file with below values:
```
/* eslint-disable */
const path = require('path');

module.exports = function override(config, env) {
  config.target = 'electron-renderer';

  config.resolve = {
    ...config.resolve,
    alias: { '~': path.resolve(__dirname, 'src') }
  };
 
  return config;
};

```
**Before update versions**
webpack version: ^4.46.2
Node.js version: 12.19.4
Electron version : 11.0.0

**After Update versions**
webpack version: ^4.46.2
Node.js version: 20.14.0
Electron version : 31.0.0

it is not working with Updated version. I have tried all possible solutions from this https://github.com/webpack/webpack/issues/10035.

None of the solutions worked for me

**Comment by alexander-akait** [comment]
> If the current behavior is a bug, please provide the steps to reproduce.

Unfortunately, without this we will not be able to help you.

Potential problem areas:
1. https://github.com/webpack/webpack/issues/10035#issuecomment-864497801
2. https://github.com/webpack/webpack/issues/10035#issuecomment-1075074282
3. Your code (it can be in dependencies) use `global` and you will need to rewrite it

If you want a real solution and explanation of the problem, you need to provide minimally reproducible code, without it, unfortunately, we cannot help, as soon as you provide additional information i will move it to issues again or show the solution

**Reply by mohan-reviewer1** [reply]
@alexander-akait  It works when contextIsolation is set to false, but we encounter a global is not defined error when contextIsolation is **true**. If we attempt to add global to the config-overrides.js file, we then get a **require is not defined error.**
![image](https://github.com/user-attachments/assets/1e5d56bb-3753-42b0-90ff-8ef3bd809fda)
![image](https://github.com/user-attachments/assets/da24c879-49bc-47f1-a483-b4de95821fe2)
I understand require module is not available for render Environment. is there any way to make it available or any suggestion for should i debug the issue?
